Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -26,44 +30,47 @@ public CategoryController(ICategoryRepository categoryRepository)
 
         [HttpPost]
         /// <summary>
-        /// Post Method 
-        /// Will Add a Catagory Name in Category Model
+        /// Api to add Category
         ///</summary>
-        /// <param name="category">Object of  class Category</param>
-        /// <returns>object of the class </returns>
-        public IActionResult CatagoryAdd([FromBody] Category category)
+        /// <param name="category">category object contains Category details</param>
+        /// <returns>Task</returns>
+        public async Task<IActionResult> AddCategory([FromBody] Category category)
         {
             if (!ModelState.IsValid)
             {
                 return BadRequest(ModelState);
             }
-            _categoryRepository.AddCategory(category);
+            if (await _categoryRepository.IsCategoryNameExistsAsync(category))
+            {
+                ModelState.AddModelError("error", _stringConstants.CategoryNameExistsError);
+                return BadRequest(ModelState);
+            }
+            await _categoryRepository.AddCategoryAsync(category);
             return Ok(category);
         }
 
         /// <summary>
-        /// Put Method
-        /// Will Edit a Existing Category from Category Table
+        /// Api to update Category
         /// </summary>
-        /// <param name="Id">Id is the primary key of Category Model</param>
-        /// <param name="catagory">Object of  class Category</param>
-        /// <returns>object of the class if key found or else it will return Bad request</returns>
+        /// <param name="id">Id of the Category to be update</param>
+        /// <param name="category">category object contains Category details</param>
+        /// <returns>Task</returns>
```

Comment:
I wrote earlier that "category object Contains Category details" and in some previous review told me to changed as Task so i did